src/ensembl/src/content/app/custom-download/components/checkbox-grid/CheckboxGrid.test.tsx

```diff
@@ -0,0 +1,145 @@
+import React from 'react';
+import { mount } from 'enzyme';
+import CheckboxGrid from './CheckboxGrid';
+import Checkbox from 'src/shared/checkbox/Checkbox';
+
+const checkboxOnChange = jest.fn();
+const gridData = {
+  default: {
+    symbol: {
+      id: 'symbol',
+      label: 'Symbol',
+      checkedStatus: false
+    },
+    id: {
+      id: 'id',
+      label: 'Gene stable ID',
+      checkedStatus: true
+    },
+    id_version: {
+      id: 'id_version',
+      label: 'Gene stable ID version',
+      checkedStatus: true
+    },
+    name: {
+      id: 'name',
+      label: 'Gene name',
+      checkedStatus: false
+    },
+    strand: {
+      id: 'strand',
+      label: 'Strand',
+      checkedStatus: false
+    },
+    start: {
+      id: 'start',
+      label: 'Gene start(bp)',
+      checkedStatus: false
+    },
+    end: {
+      id: 'end',
+      label: 'End',
+      checkedStatus: false
+    }
+  },
+  External: {
+    gencode_basic_annotation: {
+      id: 'gencode_basic_annotation',
+      label: 'GENCODE basic annotation',
+      checkedStatus: false
+    },
+    uniparc_id: {
+      id: 'uniparc_id',
+      label: 'UniParc ID',
+      checkedStatus: false
+    },
+    ncbi_id: {
+      id: 'ncbi_id',
+      label: 'NCBI gene ID',
+      checkedStatus: false
+    },
+    HGNC: {
+      id: 'HGNC',
+      label: 'HGNC symbol',
+      checkedStatus: false
+    },
+    go_domain: { id: 'go_domain', label: 'GO domain', checkedStatus: false }
+  }
+};
+
+describe('<CheckboxGrid />', () => {
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  let wrapper: any;
+  const defaultProps = {
+    gridData: gridData,
+    checkboxOnChange: checkboxOnChange
+  };
+
+  it('renders without error', () => {
+    wrapper = mount(<CheckboxGrid {...defaultProps} />);
+    expect(wrapper.find(CheckboxGrid).length).toEqual(1);
+  });
+
+  it('renders N number of checkbox based on the gridData', () => {
+    wrapper = mount(<CheckboxGrid {...defaultProps} />);
+
+    expect(wrapper.find(Checkbox).length).toEqual(12);
```

Comment:
Instead of using the magic number 12, can it be calculated from the gridData object?

Reply:
That's done